### PR TITLE
Introduce close buttons for tabs

### PR DESCRIPTION
**Pull request by christopher-l**
I don't seem to find the time to fix everything we discussed in https://github.com/daa84/neovim-gtk/pull/46 at once, so I'll just split the changes and open individual pull requests for now.

### CODE REVIEW

File: src/tabline.rs [hunk 1]
```diff
@@ -113,7 +120,33 @@ impl Tabline {
...
+                title.show();
+                close_btn.show();
+                self.tabs.append_page(&empty, Some(&label_box));
+
+                let tabs = self.tabs.clone();
+                let state_ref = Rc::clone(&self.state);
+                close_btn.connect_clicked(move |btn| {
+                    let current_label = btn
+                        .get_parent().unwrap();
+                    for i in 0..tabs.get_n_pages() {
```
Comment (daa84) [inline src/tabline.rs]: maybe here better to use https://developer.gnome.org/gtk3/stable/GtkNotebook.html#gtk-notebook-page-num ?
instead of loop.

Comment (christopher-l) [reply src/tabline.rs]: My problem is, that I don't know how to get the child widget (required by `page_num()`) when only having the label. It seems that the labels and children of GtkNotebook sit on the same level in the Gtk hierarchy. I suppose one could define a type implementing `IsA<Widget>` that holds a reference to the label and use it as child, but this was the simplest thing I could come up with. Do you know of anything that would help here?

Comment (daa84) [reply src/tabline.rs]: Ah, ok, you right, page_num is not good idea.
But maybe just use index (currenlty `_` variable) to pass to cliecked event? looks like tab index does not change event insert happens - this only change title.

Comment (christopher-l) [reply src/tabline.rs]: The problem with that is, that the index changes when tabs to the left are closed.

Comment (daa84) [reply src/tabline.rs]: ok, then this is ok

**Pull request merged**